src/library_pthread.js

```diff
@@ -469,6 +469,11 @@ var LibraryPThread = {
     pthread.worker.pthread = undefined;
   },
 
+  __emscripten_thread_cleanup: function(thread) {
```

Comment:
I wonder if we should hint the signature for this function, for example:
```suggestion
  __emscripten_thread_cleanup__sig: 'vi',
  __emscripten_thread_cleanup: function(thread) {
```

(I'm not sure when we should do that, for example `__pthread_create_js` has a signature hint, but `__emscript_init_main_thread_js` does not have a hint)

Reply:
These signatures are needed in dynamic linking when a side module takes that address of a JS function (i.e. calls it indirectly.. but not directly).   A side module should never call this function so I don't think its needed here.